Python/instrumentation.c

```diff
@@ -1197,7 +1197,7 @@ _Py_call_instrumentation_line(PyThreadState *tstate, _PyInterpreterFrame* frame,
     /* Special case sys.settrace to avoid boxing the line number,
      * only to immediately unbox it. */
     if (tools & (1 << PY_MONITORING_SYS_TRACE_ID)) {
-        if (tstate->c_tracefunc != NULL && line >= 0) {
```

Comment:
Presumably we are relying on the assert on line 1179, so dropping this seems fine.

Out of curiosity, how does the extra condition add enough cost to matter?

Reply:
I don't believe it really matters in an observable way, but I believe this piece of code was moved from somewhere else to make a fast path for `sys.settrace` and the `line >=0` just really just redundant at least after the move. The assertion is not far away from the code and we won't lose anything removing it. So little benefit but even less cost.